src/clover/optimisation/appraisal.py

```diff
@@ -0,0 +1,786 @@
+        round(grid_ghgs, 3),
+        round(kerosene_ghgs, 3),
+        round(kerosene_ghgs_mitigated, 3),
+        round(connections_ghgs, 3),
+        round(equipment_ghgs, 3),
+        round(om_ghgs, 3),
+        round(total_ghgs, 3),
+        round(total_system_ghgs, 3),
+    )
+
+
+def _simulation_financial_appraisal(
+    buffer_tank_addition: int,
+    clean_water_tank_addition: int,
+    convertor_addition: Dict[str, int],
+    diesel_addition: float,
+    finance_inputs: Dict[str, Any],
+    heat_exchanger_addition: int,
+    hot_water_tank_addition: int,
+    location: Location,
+    logger: Logger,
+    pv_addition: float,
+    pvt_addition: float,
+    simulation_results: pd.DataFrame,
+    storage_addition: float,
+    system_details: SystemDetails,
+    yearly_load_statistics: pd.DataFrame,
+) -> FinancialAppraisal:
+    """
+    Appraises the financial performance of a minigrid system.
+
+    Inputs:
+        - buffer_tank_addition:
+            The additional number of buffer tanks added this iteration.
+        - clean_water_tank_addition:
+            The additional number of clean-water tanks added this iteration.
+        - convertor_addition:
+            A mapping between convertor names and the size of each that was added to the
+            system this iteration.
+        - diesel_addition:
+            The additional diesel capacity added this iteration.
+        - finance_inputs:
+            The finance input information.
+        - heat_exchanger_addition:
+            The additional number of heat exchangers added this iteration.
+        - hot_water_tank_addition:
+            The additional number of hot-water tanks added this iteration.
+        - location:
+            The :class:`Location` being considered.
+        - logger:
+            The logger to use for the run.
+        - pv_addition:
+            The additional number of PV panels added this iteration.
+        - pvt_addition:
+            The additional number of PV-T panels added this iteration.
+        - simulation_results:
+            Outputs of Energy_System().simulation(...)
+        - storage_addition:
+            The additional number of electric storage devices (batteries) added this
+            iteration.
+        - system_details:
+            The details of this system.
+        - yearly_load_statistics:
+            The yearly electric load statistics for the system.
+
+    Outputs:
+        The financial appraisal of the system.
+
+    """
+
+    # Calculate new equipment costs (discounted)
+    equipment_costs = finance.discounted_equipment_cost(
+        buffer_tank_addition,
+        clean_water_tank_addition,
+        convertor_addition,
+        diesel_addition,
+        finance_inputs,
+        heat_exchanger_addition,
+        hot_water_tank_addition,
+        logger,
+        pv_addition,
+        pvt_addition,
+        storage_addition,
+        system_details.start_year,
+    ) + finance.independent_expenditure(
```

Comment:
Something to check/cross-reference would be to just make sure that the `independent_expenditure` is also discounted (cf. `discounted_equipment_cost`). I don't see why it wouldn't be, but I just thought of this here